core/logger/logger.go

```diff
@@ -117,38 +128,33 @@ func (l *zapLogger) withCallerSkip(skip int) Logger {
 	return &newLogger
 }
 
-func (l *zapLogger) WarnIf(err error) {
+func (l *zapLogger) WarnIf(err error, msg string) {
 	if err != nil {
-		l.Warn(err)
+		l.withCallerSkip(1).Warn(msg, "err", err)
 	}
 }
 
 func (l *zapLogger) ErrorIf(err error, optionalMsg ...string) {
 	if err != nil {
+		msg := "Error"
```

Comment:
Optional messages are a big 👎 for me, how many lines would break if we didn't allow them?

Reply:
There was some weirdness in other places that were trying to match an interface, but we can probably convert this one...